homeassistant/components/huawei_solar/entity_descriptions.py

```diff
@@ -0,0 +1,453 @@
+"""Entity descriptions for the Huawei Solar integration."""
+from dataclasses import dataclass
+
+import huawei_solar.register_names as rn
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    PERCENTAGE,
+    POWER_VOLT_AMPERE_REACTIVE,
+    POWER_WATT,
+    TEMP_CELSIUS,
+)
+from homeassistant.helpers.entity import EntityCategory
+
+
+@dataclass
+class HuaweiSolarSensorEntityDescription(SensorEntityDescription):
+    """Huawei Solar Sensor Entity."""
+
+
+# Every list in this file describes a group of entities which are related to each other.
+# The order of these lists matters, as they need to be in ascending order wrt. to their modbus-register.
+
+
+INVERTER_SENSOR_DESCRIPTIONS: list[HuaweiSolarSensorEntityDescription] = [
```

Comment:
I think it is fine to have these entity descriptions in sensor.py, but a core maintainer should confirm that. Most of the newer integration I have seen put the entity descriptions in the platform file.